File: packages/rook_sdk_apple_health/lib/src/data/mapper/data_source_authorizer_mappers.dart
import 'package:rook_sdk_apple_health/src/data/proto/protos.pb.dart';
import 'package:rook_sdk_core/rook_sdk_core.dart';

extension DataSourceAuthorizerMappers on DataSourceAuthorizerProto {
  DataSourceAuthorizer toDomain() {
    return DataSourceAuthorizer(
      dataSource: dataSource,
      authorized: authorized,
      authorizationUrl: authorizationUrlIsNull ? null : authorizationUrl,
    );
  }
}

File: packages/rook_sdk_apple_health/ios/Classes/Mapper/DataSourceAuthorizerMappers.swift
//
//  DataSourceAuthorizerMappers.swift
//  rook_sdk_apple_health
//
//  Created by Daniel Nolasco on 19/03/25.
//

import Foundation
import RookSDK

extension DataSourceAuthorizer {
    func toProto() -> DataSourceAuthorizerProto {
        if let realAuthorizationUrl = authorizationURL {
            return DataSourceAuthorizerProto.with {
                $0.dataSource = dataSource
                $0.authorized = authorized
                $0.authorizationURL = realAuthorizationUrl.absoluteString
                $0.authorizationURLIsNull = false
            }
        } else {
            return DataSourceAuthorizerProto.with {
                $0.dataSource = dataSource
                $0.authorized = authorized
                $0.authorizationURLIsNull = true
            }
        }
    }
}
